### ISSUE TITLE
Create example of a grouped table row

### ISSUE THREAD

**Issue by anthonydillon** [canonical/vanilla-framework#4100]
## Pattern to amend
Table

## Visual
![image](https://user-images.githubusercontent.com/1413534/139076786-c34519ca-943a-4055-a554-8534e7ed2436.png)
_Can be found here: https://ubuntu.com/cpu-compatibility_

## Context
We often want to group the some roles in the table around a common column (I assume the first one). We can solve this with `span-row` but then needs some border styling on the table. 

I am [proposing a "patterned" version](https://github.com/canonical-web-and-design/ubuntu.com/pull/10731/files#diff-1af0881f75e9b3c4d8a260f9498a6da0637bdb18fd7fd246adcc8fea010a3f1cR42) of this which will work with the mobile-card variant.

**Comment by lyubomir-popov** [comment]
The row borders are important, they prevent cell contents from appearing vertically grouped from a visual point of view. We actually have a solution for this in the modular table spec under the grouping scenario: https://ubuntucommunity.s3.dualstack.us-east-2.amazonaws.com/original/2X/f/f7acb578236f7ec0b6b4053d8f4ba827b856cfcc.png https://discourse.ubuntu.com/t/grouping/19857

**Comment by lyubomir-popov** [comment]
@clagom this seems ready to build to me. The version in @anthonydillon 's screenshot matches the modular table spec so nothing to add from visual design perspective. @bartaz do you agree? Anything missing that you'd like to see in the spec as it relates to this issue?

**Comment by bartaz** [comment]
If it's fine from design perspective it can go to build. I'm not sure exactly from the top of my head how we would implement that into table pattern, but it's something to figure out when building.

**Comment by clagom** [comment]
Should it be a variant of the table pattern? So that we can pick one or the other case by case.

**Comment by lyubomir-popov** [comment]
@clagom it already is,  the image [here](https://ubuntucommunity.s3.dualstack.us-east-2.amazonaws.com/original/2X/f/f7acb578236f7ec0b6b4053d8f4ba827b856cfcc.png) is from the discourse article on table grouping.

**Comment by clagom** [comment]
Yes sorry I meant a variant implementation wise, e.g. class name, so that we can set which one we want to use every time.

**Comment by bartaz** [comment]
@huwshimi If you wanna look into this, I guess for start adding an example of a table with grouped cells (via row span) would be worth to see if it renders correctly or does it need some amendments.

**Comment by bartaz** [comment]
As we noticed in the investigation in PR https://github.com/canonical/vanilla-framework/pull/4588 the row-span doesn't work well with mobile card version of the card.

Mobile card relies on having cells in each row to show in separate cards, while row-span uses a cell from one row to fill the others.
If we want to keep the row-span functionality of the table in mobile card version we need to work around that by adding some additional support for that (likely by adding some class names to hide cells).

**Comment by bartaz** [comment]
Triaging: Fixing it properly (including mobile table) is high effort. But adding an example of standard table with groupings should be easy enough to cover in v4 codebase.

For current codebase: just add an example of standard table using row span.
For next architecture: row span feature as part of the table spec, across variants.
